File: JavaScript/7js 1/countSalutes.js
/*
|--------------------------------------------------------------------------
| Count salutes
|--------------------------------------------------------------------------
|
*/

function countSalutes(hallway) {
   let res = 0;
   let right = 0;
   for (let i of hallway) {
      if (i == ">") right++;
      else if (i == "<") res += right;
   }
   return res * 2;
}


function countSalutes(hallway) {
   return hallway.replace(/>(?=(.*))|.()/g, "$1$2").replace(/[^<]/g, "").length * 2;
}


function countSalutes(hallway) {
   let l = 0;
   return hallway.split("").reduce((a, x) => {
      switch (x) {
         case ">":
            l++;
            break;
         case "<":
            a += l * 2;
      }
      return a;
   }, 0);
}


console.log(countSalutes(">--->---<--<"), 8);
console.log(countSalutes("<----<---<-->"), 0);
console.log(countSalutes(">-<->-<"), 6);

File: Python/7py 1/count_salutes.py
"""
|--------------------------------------------------------------------------
| Count salutes
|--------------------------------------------------------------------------
|
"""

def count_salutes(hallway):
   res, right = 0, 0
   for i in hallway:
      if i == ">": right += 1
      elif i == "<": res += right
   return res * 2


def count_salutes(hallway):
   return sum(hallway.count("<",i) for i,v in enumerate(hallway) if v==">")*2


def count_salutes(hallway):
   salutes = 0
   for index, c in enumerate(hallway):
      if c == ">":
         salutes += 2 * hallway[index:].count("<")
   return salutes


print(count_salutes('>--->---<--<'), 8)
print(count_salutes('<----<---<-->'), 0)
print(count_salutes('>-<->-<'), 6)